ui/pages/settings/flask/view-snap/view-snap.js

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import Button from '../../../../components/ui/button';
+import Typography from '../../../../components/ui/typography';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  TYPOGRAPHY,
+  COLORS,
+} from '../../../../helpers/constants/design-system';
+import SnapsAuthorshipPill from '../../../../components/app/flask/snaps-authorship-pill';
+import Box from '../../../../components/ui/box';
+import ToggleButton from '../../../../components/ui/toggle-button';
+import PermissionsConnectPermissionList from '../../../../components/app/permissions-connect-permission-list/permissions-connect-permission-list';
+import ConnectedSitesList from '../../../../components/app/connected-sites-list';
+import Tooltip from '../../../../components/ui/tooltip';
+import { SNAPS_LIST_ROUTE } from '../../../../helpers/constants/routes';
+import {
+  disableSnap,
+  enableSnap,
+  removeSnap,
+  removePermissionsFor,
+} from '../../../../store/actions';
+import { getSnaps, getSubjectsWithPermission } from '../../../../selectors';
+
+function ViewSnap() {
+  const t = useI18nContext();
+  const history = useHistory();
+  const location = useLocation();
+  const { pathname } = location;
+  const pathNameTail = pathname.match(/[^/]+$/u)[0];
+  const snaps = useSelector(getSnaps);
+  const snap = Object.entries(snaps)
+    .map(([_, snapState]) => snapState)
+    .find((snapState) => {
+      const decoded = decodeURIComponent(escape(window.atob(pathNameTail)));
+      return snapState.id === decoded;
+    });
+
+  useEffect(() => {
+    if (!snap) {
+      history.push(SNAPS_LIST_ROUTE);
+    }
+  }, [history, snap]);
+
+  const authorshipPillUrl = `https://npmjs.com/package/${snap?.manifest.source.location.npm.packageName}`;
+  const connectedSubjects = useSelector((state) =>
+    getSubjectsWithPermission(state, snap?.permissionName),
+  );
+  const dispatch = useDispatch();
+  const onDisconnect = (connectedOrigin, snapPermissionName) => {
+    dispatch(
+      removePermissionsFor({
+        [connectedOrigin]: [snapPermissionName],
+      }),
+    );
+  };
+  const onToggle = () => {
+    if (snap.enabled) {
+      dispatch(disableSnap(snap.id));
+    } else {
+      dispatch(enableSnap(snap.id));
+    }
+  };
+
+  if (!snap) {
+    return null;
+  }
+  return (
+    <div className="view-snap">
+      <div className="settings-page__content-row">
```

Comment:
Ermmm...kind of, the different settings pages use this class but the rule set itself is in the css file for the settings page component itself. I figured since other components were using it, it would be okay to use here as well. I can remove and just copy the styles into something like "view-snap__page-container" if that makes more sense?

Reply:
i think we should revisit the settings page css and extract the shared parts into a new component probably, but not in this PR. thanks @hmalik88